feedback/src/java/org/sakaiproject/feedback/util/RecaptchaServiceImpl.java

```
@@ -0,0 +1,95 @@
+                && !getSecretKey().isEmpty();
+    }
+
+    @Override
+    public String getPublicKey() {
+        return serverConfigurationService.getString(PUBLIC_KEY);
+    }
+
+    public String getSecretKey() {
+        return serverConfigurationService.getString(PRIVATE_KEY);
+    }
+
+    @Override
+    public boolean verify(String response) {
+        return verify(response, null);
+    }
+
+    @Override
+    public boolean verify(String response, String ip) {
+        try {
+            Unirest.setTimeouts(5000, 5000);
+            HttpRequest httpRequest = Unirest.get("https://www.google.com/recaptcha/api/siteverify")
+                    .queryString("secret", getSecretKey())
+                    .queryString("response", response);
+            if (ip != null && !ip.isEmpty()) {
+                httpRequest.queryString("remoteip", ip);
+            }
+            HttpResponse<JsonNode> jsonResponse = httpRequest.asJson();
+            int status = jsonResponse.getStatus();
+            if (status >= 200 && status < 300) {
+                JsonNode body = jsonResponse.getBody();
+                JSONObject object = body.getObject();
+                try {
+                    if (object != null) {
+                        boolean success = object.getBoolean("success");
+                        if (!success) {
+                            // Check if we're using a good secret.
+                            JSONArray jsonArray = object.getJSONArray("error-codes");
+                            for (Object aJsonArray : jsonArray) {
+                                if ("invalid-input-secret".equals(aJsonArray)) {
+                                    logger.warn("Secret currently used isn't valid.");
+                                }
+                            }
+                        }
+                        return success;
+                    }
+                } catch (JSONException e) {
+                    // Logged outside catch.
```

Something here? Empty catch doesn't look right...

---

The logging on the next line it used for both the exception and the if there is no object in the response.